LightBlue/Standalone/StandaloneAzureBlockBlob.cs

```diff
@@ -321,15 +327,23 @@ public Task UploadFromByteArrayAsync(byte[] buffer)
             return UploadFromByteArrayAsync(buffer, 0, buffer.Length);
         }
 
-        public async Task UploadFromByteArrayAsync(byte[] buffer, int index, int count)
+        public Task UploadFromByteArrayAsync(byte[] buffer, int index, int count)
         {
             if (buffer == null)
-            {
-                throw new ArgumentNullException("buffer");
-            }
+                throw new ArgumentNullException(nameof(buffer));
+
+            if (index < 0 || index >= buffer.Length)
```

Comment:
We've found this throws when calling `UploadFromByteArrayAsync` with an empty `byte[]`

Mercury appears to trigger this in at least one scenario, but currently unsure if it's unintentional, or expected and whether or not we should support it.  Still investigating.

Reply:
I can confirm Mercury does this somewhat intentionally.  Or at least expects to be able to upload empty blobs.

Reply:
I've fixed the overzealous argument validation and added a missing test.